backend/src/controllers/deck.js

```diff
@@ -0,0 +1,172 @@
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  cget: async (req, res, next) => {
+    try {
+      const decks = await deckService.findAll();
+      res.json(decks);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for POST /deck
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  post: async (req, res, next) => {
+    try {
+      const deck = await deckService.create({
+        name: req.body.name,
+        userId: req.user.id,
+      });
+      const createdDeck = await deckService.findById(deck.id);
+      res.status(201).json(createdDeck);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for GET /decks/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   * */
+  get: async (req, res, next) => {
+    try {
+      const deck = await deckService.findById(parseInt(req.params.id), {
+        include: Card,
+      });
+      if (!deck) return res.sendStatus(404);
+
+      res.json(deck);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for patch /decks/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  patch : async (req, res, next) => {
+    try {
+      const [deck] = await deckService.update(
+        { id: parseInt(req.params.id) },
+        req.body,
+      );
+      if (!deck) return res.sendStatus(404);
+      const updatedDeck = await deckService.findById(deck.id);
+      res.json(updatedDeck);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for DELETE /decks/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  delete: async (req, res, next) => {
+    try {
+      const nbRemoved = await deckService.remove({
+        id: parseInt(req.params.id),
+      });
+      res.sendStatus(nbRemoved ? 204 : 404);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for POST /decks/:id/cards
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   * */
+  addCard: async (req, res, next) => {
+    try {
+      const deck = await deckService.findById(parseInt(req.params.id), {
+        include: Card,
+      });
+      if (!deck) return res.sendStatus(404);
+      if (!req.user) return res.sendStatus(401);
+      if (req.user.id !== deck.userId) return res.sendStatus(403);
```

Comment:
```suggestion
      if (!deck) throw new Error('Deck not found', { cause: 'Not Found' });
      if (req.user.id !== deck.userId) throw new Error('This is not your deck', { cause: 'Unauthorized' });
```

Pas besoin de la 401, ta route est déjà protégé par le middleware

https://github.com/ESGI-69/challenge-node-vue/blob/51392b282c6e93d101e2305b15aa72ec5e1b5ea7/backend/src/middleware.js#L18-L20

Et pour les deux autres lignes, tu peux encore tout faire passer dans l'errorHandler.

https://github.com/ESGI-69/challenge-node-vue/blob/51392b282c6e93d101e2305b15aa72ec5e1b5ea7/backend/src/errorHandler.js#L56-L59